stmhal/timer.c

```diff
@@ -1303,12 +1303,15 @@ STATIC mp_obj_t pyb_timer_channel_callback(mp_obj_t self_in, mp_obj_t callback)
         self->callback = callback;
         uint8_t tim_id = self->timer->tim_id;
         if (tim_id == 1) {
+            __HAL_TIM_CLEAR_IT(&self->timer->tim, TIMER_IRQ_MASK(self->channel));
```

Comment:
Nice.

However, this can be simplified by moving this line before the if (since all 3 lines are the same).